sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/aggregate/HyperLogLogPlusPlusSuite.scala

```diff
@@ -175,4 +195,45 @@ class HyperLogLogPlusPlusSuite extends SparkFunSuite {
     hll.update(buffer, input)
     evaluateEstimate(hll, buffer, 1);
   }
+
+  test("SPARK-16484: Test sketch evaluation") {
+    val (hll, input, buffer) = createSketchEstimator(0.05, DoubleType)
+    List(1, 1, 2, 2, 3).foreach(i => {
+      input.setDouble(0, i)
+      hll.update(buffer, input)
+    })
+    val sketch = hll.eval(buffer).asInstanceOf[Array[Byte]]
+    val bb = ByteBuffer.wrap(sketch)
+    assert(bb.getInt() == 1)
+    assert(bb.getDouble() == 0.05)
+
+    // unclear if this is correctly reproducing round trip spark behavior via EmptyRow input
+    val approxDistinctCount = HyperLogLogPlusPlusEvalSketch(Literal(sketch)).eval(EmptyRow)
+    assert(approxDistinctCount == 3)
```

Comment:
we will need more test coverage for this feature. Can you please add corner cases including all supported input data types, NULL values, min and max values of each supported data type, any overflow conditions, and all error cases?

Reply:
I've extended the test suite to test the sketch generator/aggregators using the test harness that's used for testing the existing HLL++ functionality. I've got the sketch aggregation error cases tested via the negative tests added to the DataframeAggregationSuite. I'm open to adding more tests, though I could definitely use some guidance on what tests would be the most valuable given what's already already covered (IE I could probably build out some more tests that generate random data for each datatype, but I think that ends up testing the core HLL++ functionality more than the new sketch generators/aggregators as the generators/aggregators always operate on BinaryType data).